stl/inc/flat_set

```diff
@@ -314,17 +316,19 @@ public:
     _NODISCARD container_type extract() && noexcept(
         is_nothrow_move_constructible_v<container_type>) /* strengthened */ {
         // always clears the container (N4950 [flat.set.modifiers]/14 and [flat.multiset.modifiers]/10)
-        _Clear_guard<_Base_flat_set> _Always_clear{this};
+        _Clear_guard<container_type> _Always_clear{_STD addressof(_Mycont)};
         return _STD move(_Mycont);
     }
     void replace(container_type&& _Cont) {
         _STL_ASSERT(_Is_sorted(_Cont), _Msg_not_sorted);
-        _Clear_guard<_Base_flat_set> _Guard{this};
+        _Clear_guard<container_type> _Guard{_STD addressof(_Mycont)};
         _Mycont        = _STD move(_Cont);
         _Guard._Target = nullptr;
     }
 
-    iterator erase(iterator _Where) {
+    iterator erase(iterator _Where)
+        requires (!is_same_v<iterator, const_iterator>)

```

Comment:
It looks like this will never be satisfied; why not simply remove this overload of `erase`?

Reply:
Ideally, every method related to `iterator` can be deleted without behavior change. This includes non-const `begin/...` functions, `erase(iterator)` and non-const searching functions like `lower_bound`.
>set doesn't attempt to remove the non-const overloads of lower_bound etc., so I recommend following that pattern.

However, as talk about early on discord, this pr tries to avoid removing any of these functions directly (relying on convertibility instead of const_iter and iterator being the same type; `_Find` is removed as it's internal method). I agree that the code is unnecessarily complicated though...

(As to `erase` here, it is mimicking the current practice in `_Tree`)
https://github.com/microsoft/STL/blob/f392449fb72d1a387ac5025d508e8442914477f9/stl/inc/xtree#L1310-L1311

Reply:
After thinking about this again, I think the difference here is that most of `std::set`'s machinery is shared with `std::map`, so we don't know if `iterator` is `const_iterator`. `<flat_set>` is different because it lacks such sharing, so I think avoiding the duplicated member functions would be ultimately less squirrelly. Apologies for not thinking more clearly earlier.

Reply:
Thanks! Does this mean we can remove all methods related to non-const `iterator`, or instead a subset of these methods?